### PR TITLE
Restore old agilent infiniium cals for dc pulse cal.

### PR DESCRIPTION
**Pull request by JulianSKelly**
I would kindly ask that you not review the functions themselves harshly -- this is just restoring functionality that got lost as it wasn't in version control.

The code definitely isn't pretty, but I think it's important we open up this functionality to people to cal Z boards with the proper hardware. 

Nits about how exactly I implemented this are fair game.

### CODE REVIEW

**Comment by DanielSank** [comment]
This would be easier to review if the changes to the scope server weren't lumped with the changes to the calibration script.

**Comment by DanielSank** [comment]
With the default arguments `calibrate_dc_pulse` simply does nothing. I see no reason for the scope type to have a default value of `None`; I think you can just remove that.

File: ghzdac/calibrate.py [hunk 1]
```diff
@@ -433,6 +497,86 @@ def calibrateDCPulse(cxn,boardname,channel):
     return (datasetNumber(dataset))
 
 
+def calibrateDCPulse_infiniium(cxn, boardname, channel, intorext10mhz):
```
Comment (DanielSank) [inline ghzdac/calibrate.py]: `intorext10mhz` is really hard to read. How about `10_MHz_source` or `reference_source`?

File: ghzdac/calibrate.py [hunk 2]
```diff
@@ -274,6 +274,72 @@ def measureImpulseResponse(fpga, scope, baseline, pulse, dacoffsettime=6, pulsel
...
+    fpga.dac_run_sram(data,True)
+    if wait:
+        time.sleep(wait) #keep this long enough!! 40 sec for 4096, 20 sec for 2048
+
+    [timeAxis,y] = (scope.get_trace(SCOPECHANNEL_infiniium)) #start and stop in ns
+    timeAxis = [t*1.0e9 for t in timeAxis] #Agilent returns it in s, here we make ns
+
+
+    #kick out stuff below zero time in the trace
+    kickOutStuffBelowZero=True
```
Comment (DanielSank) [inline ghzdac/calibrate.py]: I really dislike this terminology. "Kick out" doesn't tell me what's going on. I suggest using more precise words like "filter" or "attenuate".

File: ghzdac/calibrate.py [hunk 3]
```diff
@@ -274,6 +274,72 @@ def measureImpulseResponse(fpga, scope, baseline, pulse, dacoffsettime=6, pulsel
...
+        for idx in range(len(timeAxis)):
+            t=timeAxis[idx]
+            if t['ns']<=100:
+                timeAxisTrunc.append(t)
+                yTrunc.append(y[idx])
+        timeAxis=timeAxisTrunc
+        y=yTrunc
+
+    #standard
+    starttime = (timeAxis[0])
```
Comment (DanielSank) [inline ghzdac/calibrate.py]: Inconsistent variable naming. The rest of the module seems to be camelCase.

File: ghzdac_cal/GHz_DAC_calibrate.py [hunk 4]
```diff
@@ -64,10 +68,11 @@ def calibrate_ac_pulse(fpga_name, zero_a, zero_b):
         return calibrate.calibrateACPulse(cxn, fpga_name, zero_a, zero_b)
 
 
-def calibrate_pulse(cxn, fpga_name):
+def calibrate_pulse(cxn, fpga_name, dc_scope = 'infiniium'):
```
Comment (DanielSank) [inline ghzdac_cal/GHz_DAC_calibrate.py]: PEP8 would be `dc_scope='infiniium'

**Comment by DanielSank** [comment]
For the most part this looks reasonable. I won't pretend that factoring the scope logic out into separate objects is appropriate because we simply don't have a consistent scope interface (although see my branch on this, it would be great for other people to give their 0.02.). I would like to see PEP8 fixes, better variable names and then I'll review again.

**Comment by JulianSKelly** [comment]
@DanielSank cleaned up a little.

**Comment by pomalley** [comment]
LGTM -- @DanielSank ?

**Pull request closed without merging**